src/librustc/middle/mem_categorization.rs

```diff
@@ -1196,12 +1195,11 @@ impl<'a, 'gcx, 'tcx> MemCategorizationContext<'a, 'gcx, 'tcx> {
 
         (*op)(self, cmt.clone(), pat);
 
+        // This function can be used during region checking when not all paths are fully
+        // resolved. Partially resolved paths in patterns can only legally refer to
+        // associated constants which don't require categorization.
```

Comment:
Actually, I made this comment after encountering an ICE. But maybe it was "no resolution at all" for a non-path pattern rather than a partial resolution. I'll investigate.